src/lib.rs

```diff
@@ -277,7 +295,13 @@ impl Metadata {
     /// ```
     pub fn new_from_path<S: AsRef<ffi::OsStr>>(path: S) -> Result<Metadata> {
         let mut err: *mut gexiv2::GError = ptr::null_mut();
-        let c_str_path = ffi::CString::new(path.as_ref().as_bytes()).unwrap();
+
+        let c_str_path_result = os_str_to_c_string(path);
+        if c_str_path_result.is_err() {
+            return Err(c_str_path_result.err().unwrap());
+        }
+        let c_str_path = c_str_path_result.unwrap();
```

Comment:
Is this something that a bit of `impl From<...` would allow us to replace with `?`

Reply:
I don't know what you mean with the `impl From<...` part but yes, the if construct can be shortened/replaced with the question mark operator (see newest commit).

Reply:
Awesome! I mistakenly thought we'd have to `impl From` to get the conversion to work, but I'm glad it's even easier than that.